### ISSUE TITLE
v1.19.1 Driver not mounting PV correctly on AKS 1.25.5 pod

### ISSUE THREAD

**Issue by sgtang** [kubernetes-sigs/blob-csi-driver#841]
**What happened**:

Occasionally deployments result in missing Storage Account data at a mountpoint, even though we are successfully using the same PV and PVC across several other pods and the csi-blob-node logs suggest the mount was successful:

```
GRPC request: {"staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/blob.csi.azure.com/777facbba33f918fb7926f13c94986d70d7d5692a584d91b4445b8b2c2a37be4/globalmount","target_path":"/var/lib/kubelet/pods/f9ede7cd-bbb6-462f-9956-277c4f79f9a3/volumes/kubernetes.io~csi/pv-foo-bar-blob/mount","volume_capability":{"AccessType":{"Mount":{"mount_flags":["-o allow_other","--file-cache-timeout-in-seconds=120"]}},"access_mode":{"mode":5}},"volume_context":{"containerName":"foo-bar-bundles","csi.storage.k8s.io/ephemeral":"false","csi.storage.k8s.io/pod.name":"foobar-deployment-697769ff5c-tgr7g","csi.storage.k8s.io/pod.namespace":"foobar","csi.storage.k8s.io/pod.uid":"f9ede7cd-bbb6-462f-9956-277c4f79f9a3","csi.storage.k8s.io/serviceAccount.name":"foobar-service-account"},"volume_id":"foobar-volume"}
> I0316 18:01:18.790092    8609 nodeserver.go:122] NodePublishVolume: volume foobar-volume mounting /var/lib/kubelet/plugins/kubernetes.io/csi/blob.csi.azure.com/777facbba33f918fb7926f13c94986d70d7d5692a584d91b4445b8b2c2a37be4/globalmount at /var/lib/kubelet/pods/f9ede7cd-bbb6-462f-9956-277c4f79f9a3/volumes/kubernetes.io~csi/pv-foo-bar-blob/mount with mountOptions: [bind]
> I0316 18:01:18.790124    8609 mount_linux.go:219] Mounting cmd (mount) with arguments ( -o bind /var/lib/kubelet/plugins/kubernetes.io/csi/blob.csi.azure.com/777facbba33f918fb7926f13c94986d70d7d5692a584d91b4445b8b2c2a37be4/globalmount /var/lib/kubelet/pods/f9ede7cd-bbb6-462f-9956-277c4f79f9a3/volumes/kubernetes.io~csi/pv-foo-bar-blob/mount)
> I0316 18:01:18.792278    8609 mount_linux.go:219] Mounting cmd (mount) with arguments ( -o bind,remount /var/lib/kubelet/plugins/kubernetes.io/csi/blob.csi.azure.com/777facbba33f918fb7926f13c94986d70d7d5692a584d91b4445b8b2c2a37be4/globalmount /var/lib/kubelet/pods/f9ede7cd-bbb6-462f-9956-277c4f79f9a3/volumes/kubernetes.io~csi/pv-foo-bar-blob/mount)
> I0316 18:01:18.794089    8609 nodeserver.go:138] NodePublishVolume: volume foobar-volume mount /var/lib/kubelet/plugins/kubernetes.io/csi/blob.csi.azure.com/777facbba33f918fb7926f13c94986d70d7d5692a584d91b4445b8b2c2a37be4/globalmount at /var/lib/kubelet/pods/f9ede7cd-bbb6-462f-9956-277c4f79f9a3/volumes/kubernetes.io~csi/pv-foo-bar-blob/mount successfully
```

The mount is listed alongside the others, but with /dev/sdb1 listed as its name and the filesystem type is ext4 instead of fuse:

```
blobfuse2 on /var/lib/kubelet/pods/1c9f2298-071c-4837-845e-8c3376595ee4/volumes/kubernetes.io~csi/pv-foo-bar-blob/mount type fuse (rw,relatime,user_id=0,group_id=0,allow_other)
blobfuse2 on /var/lib/kubelet/pods/2612d7e1-aa04-452a-bc11-1e66073dfa2b/volumes/kubernetes.io~csi/pv-foo-bar-blob/mount type fuse (rw,relatime,user_id=0,group_id=0,allow_other)
/dev/sdb1 on /var/lib/kubelet/pods/f9ede7cd-bbb6-462f-9956-277c4f79f9a3/volumes/kubernetes.io~csi/pv-foo-bar-blob/mount type ext4 (rw,relatime,discard,errors=remount-ro)
```

**What you expected to happen**:

Expected the mount to have the same filesystem name and type as the other successful mounts, OR expected an error to indicate why this mount isn't functioning/exposing data.

**How to reproduce it**:
It happens intermittently from our current perspective. We did just upgrade to Kubernetes v1.25.5 a couple of weeks ago.

**Environment**:
- CSI Driver version: mcr.microsoft.com/oss/kubernetes-csi/blob-csi:v1.19.1
- Kubernetes version (use `kubectl version`): 1.25.5
- OS (e.g. from /etc/os-release): Node: NAME="Debian GNU/Linux"
VERSION_ID="11", Pod: NAME="Ubuntu" VERSION="20.04.3 LTS (Focal Fossa)"
- Kernel (e.g. `uname -a`): 5.15.0-1033-azure

**Comment by andyzhangx** [comment]
related to https://github.com/Azure/azure-storage-fuse/issues/1079, still in investigating

**Comment by andyzhangx** [comment]
update: we are going to fix this issue in blob csi driver `v1.19.2` and `v1.20.1`, and then rollout the fixed version on AKS, if you have any AKS cluster hitting this issue, pls email me the aks cluster api-server address (you could find my mail address in my github account), I would help mitigate this issue immediately so you don't need to wait a few days for AKS release roll out in your region.

**Comment by sgtang** [comment]
Thanks for the quick turnaround! We will wait a few days for the managed upgrade since this is mainly an issue in our dev cluster currently.
